Regular-Question-Answers/group-anagrams.cpp

```diff
@@ -0,0 +1,47 @@
+//Author:Ömer Faruk ERDEM
+//Question Link = https://leetcode.com/problems/group-anagrams/
+class Solution {
+public:
+
+    /* Function to return anagram groups 
+     * Given string vector as an input and function will return the anagrams according their group
+     * So ,what do we mean by anagram group:
+     * For example if there are words contains same letters but different order then they are anagram.
+     * And  today we will group them according to their characters
+     */
+    vector<vector<string>> groupAnagrams(vector<string>& strs) {
+        // First create a two dimensional result vector to store anagram groups
+        vector<vector<string>> result; 
+
+        /* Now the thing ı am doing here is very important because here ı create a map with pair .
+         * First element of the pair contains string and second element of the pair contains vector of strings .
+         *  That allows us to store the strings that has same characters but different order.
+         */
+        unordered_map< string , vector<string> >  mp;

```

Comment:
unordered_map<string, vector<string>> mp;

Reply:
I think that kinf of creating map doesn't work.

Reply:
I think it is not working because of your compiler, if you that on leetcode I am sure it will work. On the other hand, on your computer, you can change to use g++-11 instead of g++ compiler by doing that some problems such as this (or #include<bits/stdc++.h> library and etc.) will be fixed.